### ISSUE TITLE
Compendium Browser not showing

### ISSUE THREAD

**Issue by ravingdragoon** [League-of-Foundry-Developers/compendium-browser#25]
Since updating to 0.6.0 I am receiving an error in the console (see below) and I am not seeing the compendium browser button in the Compendiums Sidebar. However, do see the Module under Module Settings.

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'dnd5e.items')  
    at CompendiumBrowser.initSettings (compendium-browser.js:1249)  
    at CompendiumBrowser.initialize (compendium-browser.js:72)  
    at compendium-browser.js:1512  
    at Function._call (eval at <anonymous> (assets.forge-vtt.com/bazaar/modules/lib-wrapper/1.10.4.0/lib-wrapper.js:1), <anonymous>:4:14)  
    at Function.callAll (foundry.js:153)  
    at Game.setupGame (foundry.js:4403)  
    at async Game._initializeGameView (foundry.js:5213)  
    at async Game.initialize (foundry.js:4286)  
  initSettings @ compendium-browser.js:1249
  initialize @ compendium-browser.js:72
  (anonymous) @ compendium-browser.js:1512
  _call @ VM42266:4
  callAll @ foundry.js:153
  setupGame @ foundry.js:4403

I receive the error whether Compendium Browser is on with other modules or if it is the only module active. I am running Foundry 0.8.9, on The Forge, in Chrome Version 93.0.4577.63 (Official Build) (64-bit)

**Comment by spetzel2020** [comment]
Thanks for the quick report! A couple of questions:
1. Are you using CB with dnd5e or another system?
2. I assume you've successfully used it previously based on your comment
about upgrading?
I'm traveling rn but will aim to look at it this evening or tomorrow.

**Comment by ravingdragoon** [comment]
Yes, I am using it with dnd5e and I have successfully used it previously.

**Comment by spetzel2020** [comment]
Hello, I have tested this evening against Foundry 0.8.9 on the Forge and cannot reproduce this error. 

However, looking at the console error you report I think it's  a corrupted item within dnd5e.items, or maybe not finding dnd5e.items at all - is it present still? What version of dnd5e are you running? Is there any chance that the dnd5e upgrade failed - I don't know how to force a reinstall of dnd5e but I suspect there is a way (or at least of the compendia).

Let me know, especially re dnd5e.items being present. In my version , it has 820 items - is that what you have?

That all said, CB should recover more gracefully from a problem like this.

**Comment by ravingdragoon** [comment]
I just updated to 5e 1.4.3 and I am still getting the error. Using Compendium Folders' fix compendium tool it says there is nothing wrong with my dnd5e.items compendium.

**Comment by spetzel2020** [comment]
Ok, I may need to instrument a version for you that will help diagnose this.

**Comment by spetzel2020** [comment]
Looking more closely at the code, it suggests something is borked in your saved settings for Compendium Browser. If you feel comfortable doing so, can you open the console and type in this and paste what you get. 
`game.settings.get("compendium-browser", 'settings');`

You can also DM me on Discord at Spetzel#0103 if that's easier.

**Comment by ravingdragoon** [comment]
Sure, here it is:

game.settings.get("compendium-browser", 'settings');  
{loadedCompendium: {…}, allowSpellBrowser: true, allowFeatBrowser: true, allowItemBrowser: true, allowNpcBrowser: false}  
allowFeatBrowser: true  
allowItemBrowser: true  
allowNpcBrowser: false  
allowSpellBrowser: true  
loadedCompendium:  
Actor:  
ActiveAuras.Active Auras Zone Actors: {load: true, name: 'Active Auras Zone Actors (ActiveAuras.Active Auras Zone Actors)'}  
automated-evocations.aeactors: {load: true, name: 'Automated Evocations Summons (automated-evocations.aeactors)'}  
ddd-beginner-baubles.monsters: {load: true, name: 'Monsters (Beginner Baubles) (ddd-beginner-baubles.monsters)'}  
dnd5e.heroes: {load: true, name: 'Starter Heroes (dnd5e.heroes)'}  
dnd5e.monsters: {load: true, name: 'Monsters (SRD) (dnd5e.monsters)'}  
jb2a_patreon.jb2a-actors: {load: true, name: 'JB2A Actors (jb2a_patreon.jb2a-actors)'}  
trigger-happy.actors: {load: true, name: 'Trigger Actors (trigger-happy.actors)'}  
world.ddb-curse-of-strahd-backup-monsters: {load: true, name: 'DDB Monsters (world.ddb-curse-of-strahd-backup-monsters)'}  
[[Prototype]]: Object  
Item:  
ATL.Token Lighting Premade: {load: true, name: 'Token Lighting premade (ATL.Token Lighting Premade)'}  
ActiveAuras.Active Auras Auras: {load: true, name: 'Active Auras Auras (ActiveAuras.Active Auras Auras)'}  
Dynamic-Effects-SRD.DAE SRD Feats: {load: true, name: 'DAE SRD Feats (Dynamic-Effects-SRD.DAE SRD Feats)'}  
Dynamic-Effects-SRD.DAE SRD Items: {load: true, name: 'DAE SRD  Items (Dynamic-Effects-SRD.DAE SRD Items)'}  
Dynamic-Effects-SRD.DAE SRD Magic Items: {load: true, name: 'SRD Magic Items (Dynamic-Effects-SRD.DAE SRD Magic Items)'}  
My-Shared-Compendia.Classes: {load: true, name: 'DDB Classes (My-Shared-Compendia.Classes)'}  
My-Shared-Compendia.Feats: {load: true, name: 'DDB Feats (My-Shared-Compendia.Feats)'}  
My-Shared-Compendia.Override: {load: true, name: 'DDB Override (My-Shared-Compendia.Override)'}  
My-Shared-Compendia.Races: {load: true, name: 'DDB Races (My-Shared-Compendia.Races)'}  
My-Shared-Compendia.Racial Traits: {load: true, name: 'DDB Racial Traits (My-Shared-Compendia.Racial Traits)'}  
dae.premadeitems: {load: true, name: 'Premade Items (dae.premadeitems)'}  
ddd-beginner-baubles.items: {load: true, name: 'Items (Beginner Baubles) (ddd-beginner-baubles.items)'}  
dnd5e.classes: {load: true, name: 'Classes (SRD) (dnd5e.classes)'}  
dnd5e.classfeatures: {load: true, name: 'Class Features (SRD) (dnd5e.classfeatures)'}  
dnd5e.items: {load: true, name: 'Items (SRD) (dnd5e.items)'}  
dnd5e.monsterfeatures: {load: true, name: 'Monster Features (SRD) (dnd5e.monsterfeatures)'}  
dnd5e.races: {load: true, name: 'Racial Features (SRD) (dnd5e.races)'}  
dnd5e.spells: {load: true, name: 'Spells (SRD) (dnd5e.spells)'}  
dnd5e.tradegoods: {load: true, name: 'Trade Goods (SRD) (dnd5e.tradegoods)'}  
giffyglyphs-5e-monster-maker.gmm-monster-attacks: {load: true, name: 'GMM Monster Attacks (giffyglyphs-5e-monster-maker.gmm-monster-attacks)'}  
giffyglyphs-5e-monster-maker.gmm-monster-powers: {load: true, name: 'GMM Monster Powers (giffyglyphs-5e-monster-maker.gmm-monster-powers)'}  
innocenti-looting.looting-items: {load: true, name: 'Looting Items Currency (innocenti-looting.looting-items)'}  
itemcollection.packs: {load: true, name: 'Premade Bags (itemcollection.packs)'}  
magicitems.magicitem-feats: {load: true, name: 'Magic Items: Item Feats (magicitems.magicitem-feats)'}  
magicitems.magicitems: {load: true, name: 'Magic Items (magicitems.magicitems)'}  
midi-qol.midiqol-sample-items: {load: true, name: 'MidiQOL Sample Items (midi-qol.midiqol-sample-items)'}  
midi-srd.Midi SRD Feats: {load: true, name: 'Midi SRD Feats (midi-srd.Midi SRD Feats)'}  
midi-srd.Midi SRD Items: {load: true, name: 'Midi SRD  Items (midi-srd.Midi SRD Items)'}  
midi-srd.Midi SRD Spells: {load: true, name: 'Midi SRD Spells (midi-srd.Midi SRD Spells)'}  
world.ddb-curse-of-strahd-backup-class-features: {load: true, name: 'DDB Class Features (world.ddb-curse-of-strahd-backup-class-features)'}  
world.ddb-curse-of-strahd-backup-classes: {load: true, name: 'DDB Classes (world.ddb-curse-of-strahd-backup-classes)'}  
world.ddb-curse-of-strahd-backup-feats: {load: true, name: 'DDB Feats (world.ddb-curse-of-strahd-backup-feats)'}  
world.ddb-curse-of-strahd-backup-items: {load: true, name: 'DDB Items (world.ddb-curse-of-strahd-backup-items)'}  
world.ddb-curse-of-strahd-backup-override: {load: true, name: 'DDB Override (world.ddb-curse-of-strahd-backup-override)'}  
world.ddb-curse-of-strahd-backup-races: {load: true, name: 'DDB Races (world.ddb-curse-of-strahd-backup-races)'}  
world.ddb-curse-of-strahd-backup-racial-traits: {load: true, name: 'DDB Racial Traits (world.ddb-curse-of-strahd-backup-racial-traits)'}  
world.ddb-curse-of-strahd-backup-spells: {load: true, name: 'DDB Spells (world.ddb-curse-of-strahd-backup-spells)'}  
[[Prototype]]: Object  
[[Prototype]]: Object  
[[Prototype]]: Object  
constructor: ƒ Object()  
hasOwnProperty: ƒ hasOwnProperty()  
isPrototypeOf: ƒ isPrototypeOf()  
propertyIsEnumerable: ƒ propertyIsEnumerable()  
toLocaleString: ƒ toLocaleString()  
toString: ƒ toString()  
valueOf: ƒ valueOf()  
__defineGetter__: ƒ __defineGetter__()  
__defineSetter__: ƒ __defineSetter__()  
__lookupGetter__: ƒ __lookupGetter__()  
__lookupSetter__: ƒ __lookupSetter__()  
__proto__: (...)  
get __proto__: ƒ __proto__()  
set __proto__: ƒ __proto__()

**Comment by spetzel2020** [comment]
Interesting. This is what I have:
```
{
    "loadedSpellCompendium": {
        "dnd5e.items": {
            "load": true,
            "name": "Items (SRD) (dnd5e.items)"
        },
        "dnd5e.tradegoods": {
            "load": true,
            "name": "Trade Goods (SRD) (dnd5e.tradegoods)"
        },
        "dnd5e.spells": {
            "load": false,
            "name": "Spells (SRD) (dnd5e.spells)"
        },
        "dnd5e.classes": {
            "load": true,
            "name": "Classes (SRD) (dnd5e.classes)"
        },
        "dnd5e.classfeatures": {
            "load": true,
            "name": "Class Features (SRD) (dnd5e.classfeatures)"
        },
        "dnd5e.races": {
            "load": false,
            "name": "Racial Features (SRD) (dnd5e.races)"
        },
        "dnd5e.monsterfeatures": {
            "load": true,
            "name": "Monster Features (SRD) (dnd5e.monsterfeatures)"
        },
        "magicitems.magicitems": {
            "load": false,
            "name": "Magic Items (magicitems.magicitems)"
        },
        "magicitems.magicitem-feats": {
            "load": true,
            "name": "Magic Items: Item Feats (magicitems.magicitem-feats)"
        },
        "world.ddb-zamin-spells": {
            "load": true,
            "name": "DDB Spells (world.ddb-zamin-spells)"
        },
        "world.ddb-zamin-items": {
            "load": false,
            "name": "DDB Items (world.ddb-zamin-items)"
        },
        "world.ddb-zamin-class-features": {
            "load": true,
            "name": "DDB Class Features (world.ddb-zamin-class-features)"
        },
        "world.ddb-zamin-classes": {
            "load": true,
            "name": "DDB Classes (world.ddb-zamin-classes)"
        },
        "world.ddb-zamin-racial-traits": {
            "load": true,
            "name": "DDB Racial Traits (world.ddb-zamin-racial-traits)"
        },
        "world.ddb-zamin-feats": {
            "load": true,
            "name": "DDB Feats (world.ddb-zamin-feats)"
        },
        "world.ddb-zamin-races": {
            "load": true,
            "name": "DDB Races (world.ddb-zamin-races)"
        },
        "world.ddb-zamin-override": {
            "load": true,
            "name": "DDB Override (world.ddb-zamin-override)"
        }
    },
    "loadedNpcCompendium": {
        "dnd5e.heroes": {
            "load": true,
            "name": "Starter Heroes (dnd5e.heroes)"
        },
        "dnd5e.monsters": {
            "load": true,
            "name": "Monsters (SRD) (dnd5e.monsters)"
        },
        "world.ddb-zamin-monsters": {
            "load": true,
            "name": "DDB Monsters (world.ddb-zamin-monsters)"
        }
    },
    "allowSpellBrowser": false,
    "allowFeatBrowser": false,
    "allowItemBrowser": false,
    "allowNpcBrowser": false
}
```

**Comment by spetzel2020** [comment]
Fixed in 0.6.1 and 0.7.1; release coming soon.
